### ISSUE TITLE
File#pos fails for files greater than java.lang.Integer::MAX_VALUE on 64-bit systems

### ISSUE THREAD

**Issue by byteit101** [jruby/jruby#3664]
When writing files greater than 2GiB, the Tempfile#pos value throws an error.
```rb
require 'tempfile'
halfG = "12345" * 100_000_000
tmp = Tempfile.new("demo")
9.times do
	puts tmp.write(halfG)
	begin
		puts tmp.pos # will error once > 2GiB
	rescue
		puts "Error, can still write"
		puts tmp.write(halfG)
		puts tmp.size # works fine
		puts tmp.pos # will error
	end
end
```

JRuby 9.0.5.0 (2G heap):
````
$ ruby -J-Xmx2000m bug-demo.rb
500000000
500000000
500000000
1000000000
500000000
1500000000
500000000
2000000000
500000000
Error, can still write
500000000
3000000000
Errno::ENOENT: No such file or directory - /tmp/demo20160213-14805-1qqz35b
                   pos at org/jruby/RubyIO.java:1532
  block in bug-demo.rb at bug-demo.rb:13
                 times at org/jruby/RubyFixnum.java:296
                 <top> at bug-demo.rb:5
```

MRI:
```
$ ruby bug-demo.rb 
500000000
500000000
500000000
1000000000
500000000
1500000000
500000000
2000000000
500000000
2500000000
500000000
3000000000
500000000
3500000000
500000000
4000000000
500000000
4500000000
```

**Comment by byteit101** [comment]
Also just realized that this is in File#pos too, not just Tempfile.
```
tmp = File.new("/tmp/demo", "w")
```
(other code is the same)

**Comment by byteit101** [comment]
Digging in, this looks to be a jnr-posix issue as all the declarations are ints, not longs. Should I cross-post this issue in jnr?

https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/util/io/PosixShim.java#L69
https://github.com/jnr/jnr-posix/blob/master/src/main/java/jnr/posix/POSIX.java#L153

**Comment by headius** [comment]
@byteit101 Yeah we'll need to fix those APIs to be longs. Anything that's using NIO channels only should be long already.

**Comment by byteit101** [comment]
I just tried this on a 32-bit system, and it worked as expected, which I did not expect

**Comment by headius** [comment]
See #3817 which aggregates ongoing work to fix seek/tell/pos logic.

Some of the errors here will be fixed along with #3435, but the remaining API fixes still need to be done.

**Comment by headius** [comment]
See #3817 which aggregates ongoing work to fix seek/tell/pos logic.

Some of the errors here will be fixed along with #3435.
